src/main/java/base/AnnotationVisitor.java

```diff
@@ -30,41 +33,67 @@ public AnnotationVisitor() {
     @Override
     public void visit(
-        MarkerAnnotationExpr annotationExpr, Void args) {
+        NormalAnnotationExpr annotationExpr, Void args) {
         super.visit(annotationExpr, args);
 
-        if (isPatternAnnotation(annotationExpr)) {
-            Optional<CompilationUnit> optional =
-                annotationExpr.findRootNode().findCompilationUnit();
-            if (optional.isEmpty()) {
-                return;
-            }
-            CompilationUnit compUnit = optional.get();
+        Optional<CompilationUnit> optional = annotationExpr.findRootNode().findCompilationUnit();
+        if (optional.isEmpty()) {
+            return;
+        }
 
-            Pattern pattern = Pattern.valueOf(
-                annotationExpr.getNameAsString().toUpperCase(Locale.ENGLISH));
+        CompilationUnit compilationUnit = optional.get();
 
-            PatternVerifierFactory.getVerifier(pattern).verify(compUnit);
+        for (Pattern pattern : getPatternsFromAnnotation(annotationExpr)) {
+
+            boolean validPattern = PatternVerifierFactory.getVerifier(pattern).verify(
+                compilationUnit);
+            // Everything below this is just to show something, will probably
+            // not be here in the future
+            String fileName;
+            if (compilationUnit.getStorage().isEmpty()) {
+                fileName = "File name not found";
+            } else {
+                fileName = compilationUnit.getStorage().get().getFileName();
+            }
+
+            System.out.println(
+                "File: " + fileName + "\nTested patterns:\n" + pattern + ": " + validPattern);
         }
     }
 
-    /**
-     * Identifies whether or not a given annotation is associated with a known design pattern, which
-     * and therefore indicates that said pattern should exist.
-     *
-     * @param ann the annotation to verify
-     *
-     * @return true if {@link Pattern} contains the given annotation
-     */
-    private boolean isPatternAnnotation(AnnotationExpr ann) {
-        Pattern[] patterns = Pattern.class.getEnumConstants();
-        for (Pattern p : patterns) {
-            if (ann.getNameAsString().equalsIgnoreCase(p.toString())) {
-                return true;
+    private List<Pattern> getPatternsFromAnnotation(NormalAnnotationExpr annotation) {
+        NodeList<MemberValuePair> pairs = annotation.getPairs();
+        List<Pattern> patterns = new ArrayList<>();
+
+        for (int i = 0; i < pairs.size(); i++) {
+            MemberValuePair pair = pairs.get(i);
+            if (isPatternKey(pair)) {
+                var patternsInPair = pair.getValue().asArrayInitializerExpr().getValues().toArray();
+                for (int j = 0; j < patternsInPair.length; j++) {
+                    for (Pattern pattern : Pattern.values()) {
+                        if (isDesignPatternEnum(
+                            patternsInPair[i].toString(), pattern, "pattern.")) {
```

Comment:
```suggestion
                             patternsInPair[j].toString(), pattern, "pattern.")) {
```
(or use short for loop to avoid this problem altogether)

Reply:
man kan anta att det fungerar men isPatternKey säkerställer att vi kollar på rätt MemberValuePair då det kan finnas flera